pxr/usdImaging/usdImaging/instanceAdapter.cpp

```diff
@@ -2674,7 +2674,7 @@ UsdImagingInstanceAdapter::GetScenePrimPaths(
                     }
                 }
 
-                result.resize(validIndices);
+                result.resize(instanceIndices.size());
```

Comment:
If my understanding of the PR description is correct the result vector has to have the correct size always and thus it has to be resized earlier. Good places are line 2638 or line 2657. Is 0 a good default when passing invalid indices?

Reply:
You mean to match the result size with the inputIndices size even when there are no valid indices? That's a good point. I suspect it is my commit message that is erroneous here, and returning a completely empty result vector is probably okay too in this situation? Though I'm also happy to move the resize call up higher if there is general agreement that the output vector size should match the input vector size always.

I'm not sure what you mean by "Is 0 a good default when passing invalid indices?" Is zero a good default for what?

Reply:
Yes, if the result vector must match the size of the input indices this should be true even if all indices are invalid.

>I'm not sure what you mean by "Is 0 a good default when passing invalid indices?" Is zero a good default for what?
Was was under the wrong impression that result contains integer value. Since it contains paths the default value is good and the result should be resized before valid indices.